ReaderAttributes.h does not include Guid.h

I'm just getting a project set up, and encountered the following build error.

```
liam@drungus fc (master #)$ make
c++ -std=c++11 -MD -MP -Os -ggdb -fPIC -fno-common -fvisibility=hidden -Wall -Werror -Wundef -I./vendor/Fast-RTPS-v1.0.0-1415-g7b323d1/build/install/include -c src/listener.cpp -o src/listener.o
In file included from src/listener.cpp:1:
In file included from src/listener.h:7:
In file included from ./vendor/Fast-RTPS-v1.0.0-1415-g7b323d1/build/install/include/fastrtps/attributes/SubscriberAttributes.h:19:
./vendor/Fast-RTPS-v1.0.0-1415-g7b323d1/build/install/include/fastrtps/attributes/../rtps/attributes/ReaderAttributes.h:83:2: error: unknown type name 'GUID_t'
        GUID_t guid;
        ^
1 error generated.
make: *** [src/listener.o] Error 1
```

If `ReaderAttributes.h` requires `GUID_t`, it seems like it should include `Guid.h`.

I worked around it for now by adding `#include "fastrtps/rtps/common/all_common.h"` before I include `ReaderAttributes.h`

--- a/include/fastrtps/TopicDataType.h
+++ b/include/fastrtps/TopicDataType.h
@@ -85,8 +85,8 @@ public:
 	* Get topic data type name
 	* @return Topic data type name
 	*/
-	RTPS_DllAPI inline const char* getName(){ return m_topicDataTypeName.c_str(); }
-	
+	RTPS_DllAPI inline const char* getName() const { return m_topicDataTypeName.c_str(); }
+
 	//! Maximum Type size in bytes. (If the type includes a string the user MUST ensure that the maximum
 	//! size of the string respect the maximum defined size.).
 	uint32_t m_typeSize;

--- a/include/fastrtps/rtps/attributes/RTPSParticipantAttributes.h
+++ b/include/fastrtps/rtps/attributes/RTPSParticipantAttributes.h
@@ -217,7 +217,7 @@ public:
 	//!Set the name of the participant.
 	inline void setName(const char* nam){name = nam;}
 	//!Get the name of the participant.
-	inline const char* getName(){return name.c_str();}
+	inline const char* getName() const {return name.c_str();}
 
 private:
 	//!Name of the participant.

--- a/include/fastrtps/rtps/attributes/ReaderAttributes.h
+++ b/include/fastrtps/rtps/attributes/ReaderAttributes.h
@@ -7,7 +7,7 @@
  *************************************************************************/
 
 /**
- * @file WriterAttributes.h
+ * @file ReaderAttributes.h
  *
  */
 
@@ -15,6 +15,7 @@
 #define READERATTRIBUTES_H_
 
 #include "../common/Time_t.h"
+#include "../common/Guid.h"
 #include "EndpointAttributes.h"
 namespace eprosima{
 namespace fastrtps{

